spec/unit/puppet-strings/json_spec.rb

```diff
@@ -27,8 +27,8 @@ class klass(Integer $param1, $param2, String $param3 = hi) inherits foo::bar {
 # @param param1 First param.
 # @param param2 Second param.
 # @param param3 Third param.
-# @return [Undef] Returns nothing.
-function func(Integer $param1, $param2, String $param3 = hi) {
+# @return Returns nothing.
+function func(Integer $param1, $param2, String $param3 = hi) >> Undef {
```

Comment:
Added a complex example to the function handler and parser spec, but left the json test alone. I think the JSON test is starting to go beyond the scope of what it should do, which is just ensure JSON output works and matches our schema.